activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/model/ProcessExtensionModel.java

```diff
@@ -13,21 +13,32 @@
 
 package org.activiti.spring.process.model;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class ProcessExtensionModel {
 
     private String id;
-    private Extension extensions = new Extension();
+    private Map<String, Extension> extensions = new HashMap();
 
     public String getId() {
         return id;
     }
+
     public void setId(String id) {
         this.id = id;
     }
-    public Extension getExtensions() {
-        return extensions;
+
+    public Extension getExtensions(String processDefinitionKey) {
+        return extensions.get(processDefinitionKey);
+    }
+
+    public Extension getExtensionsByProcessDefinitionId(String processDefinitionId) {
+        String processDefinitionKey = processDefinitionId.split(":")[0];
```

Comment:
Do not make this assumption, I'm not sure that this is true all the time.

Reply:
When this is not true?

Reply:
The good news is that after the refactoring this code is not needed anymore!